pkg/machine/e2e/init_test.go

```diff
@@ -363,6 +363,41 @@ var _ = Describe("podman machine init", func() {
 		Expect(err).ToNot(HaveOccurred())
 		Expect(inspectShouldPass).To(Exit(0))
 	})
+
+	It("machine init with rosetta", func() {
+		if testProvider.VMType() != define.AppleHvVirt {
+			Skip("Test is only for AppleHv")
```

Comment:
Can you check if the VMType != applehvvirt, that it returns an error?

Reply:
Test results for WSL and Hyper-V.
```
> $env:CONTAINERS_MACHINE_PROVIDER="wsl"
> .\winmake.ps1 localmachine "init_test.go"
------------------------------
podman machine init
C:/Users/user/podman/pkg/machine/e2e/init_test.go:20
  machine init with rosetta
  C:/Users/user/podman/pkg/machine/e2e/init_test.go:367
  > Enter [BeforeEach] podman machine init - C:/Users/user/podman/pkg/machine/e2e/init_test.go:26 @ 02/16/24 02:36:09.168
--> copying "C:\\Users\\user\\AppData\\Local\\Temp\\rootfs.tar" to "C:\\Users\\user\\AppData\\Local\\Temp\\podman_test75595674\\rootfs.tar"/n  < Exit [BeforeEach] podman machine init - C:/Users/user/podman/pkg/machine/e2e/init_test.go:26 @ 02/16/24 02:36:09.472 (305ms)
  > Enter [It] machine init with rosetta - C:/Users/user/podman/pkg/machine/e2e/init_test.go:367 @ 02/16/24 02:36:09.477
  [SKIPPED] Test is only for AppleHv
  In [It] at: C:/Users/user/podman/pkg/machine/e2e/init_test.go:369 @ 02/16/24 02:36:09.482

  Full Stack Trace
    github.com/containers/podman/v5/pkg/machine/e2e_test.init.func4.13()
        C:/Users/user/podman/pkg/machine/e2e/init_test.go:369 +0x65
  < Exit [It] machine init with rosetta - C:/Users/user/podman/pkg/machine/e2e/init_test.go:367 @ 02/16/24 02:36:09.494 (16ms)
  > Enter [AfterEach] podman machine init - C:/Users/user/podman/pkg/machine/e2e/init_test.go:29 @ 02/16/24 02:36:09.499
  < Exit [AfterEach] podman machine init - C:/Users/user/podman/pkg/machine/e2e/init_test.go:29 @ 02/16/24 02:36:09.549 (50ms)
S [SKIPPED] [0.390 seconds]
podman machine init
C:/Users/user/podman/pkg/machine/e2e/init_test.go:20
  [It] machine init with rosetta
  C:/Users/user/podman/pkg/machine/e2e/init_test.go:367
------------------------------

> $env:CONTAINERS_MACHINE_PROVIDER="hyperv"
> .\winmake.ps1 localmachine "init_test.go"
------------------------------
podman machine init
C:/Users/user/podman/pkg/machine/e2e/init_test.go:20
  machine init with rosetta
  C:/Users/user/podman/pkg/machine/e2e/init_test.go:367
  > Enter [BeforeEach] podman machine init - C:/Users/user/podman/pkg/machine/e2e/init_test.go:26 @ 02/16/24 02:46:15.016
--> copying "C:\\Users\\user\\AppData\\Local\\Temp\\fedora-coreos-39.20240210.2.0-hyperv.x86_64.vhdx" to "C:\\Users\\user\\AppData\\Local\\Temp\\podman_test1050276269\\fedora-coreos-39.20240210.2.0-hyperv.x86_64.vhdx"/n  < Exit [BeforeEach] podman machine init - C:/Users/user/podman/pkg/machine/e2e/init_test.go:26 @ 02/16/24 02:46:16.22 (1.204s)
  > Enter [It] machine init with rosetta - C:/Users/user/podman/pkg/machine/e2e/init_test.go:367 @ 02/16/24 02:46:16.226
  [SKIPPED] Test is only for AppleHv
  In [It] at: C:/Users/user/podman/pkg/machine/e2e/init_test.go:369 @ 02/16/24 02:46:16.231

  Full Stack Trace
    github.com/containers/podman/v5/pkg/machine/e2e_test.init.func4.13()
        C:/Users/user/podman/pkg/machine/e2e/init_test.go:369 +0x65
  < Exit [It] machine init with rosetta - C:/Users/user/podman/pkg/machine/e2e/init_test.go:367 @ 02/16/24 02:46:16.242 (16ms)
  > Enter [AfterEach] podman machine init - C:/Users/user/podman/pkg/machine/e2e/init_test.go:29 @ 02/16/24 02:46:16.246
  < Exit [AfterEach] podman machine init - C:/Users/user/podman/pkg/machine/e2e/init_test.go:29 @ 02/16/24 02:46:16.393 (148ms)
S [SKIPPED] [1.382 seconds]
podman machine init
C:/Users/user/podman/pkg/machine/e2e/init_test.go:20
  [It] machine init with rosetta
  C:/Users/user/podman/pkg/machine/e2e/init_test.go:367
------------------------------
```